**Laboratorio 5**

María Paula Angulo Collantes

Danna Gabriela Castro

In [12]:
import pandas as pd
import numpy as np

In [13]:
df = pd.read_csv("anage_animalia.csv")

**Punto 1**

In [25]:
print("Dimensiones:", df.shape)

print("\nTipos de variables:")
print(df.dtypes)

print("\nValores faltantes:")
print(df.isnull().sum())

print("\nData quality:")
print(df["Data quality"].value_counts())

print("\nSpecimen origin:")
print(df["Specimen origin"].value_counts())

print("\nSample size:")
print(df["Sample size"].describe())

Dimensiones: (4513, 31)

Tipos de variables:
HAGRID                                int64
Kingdom                                 str
Phylum                                  str
Class                                   str
Order                                   str
Family                                  str
Genus                                   str
Species                                 str
Common name                             str
Female maturity (days)              float64
Male maturity (days)                float64
Gestation/Incubation (days)         float64
Weaning (days)                      float64
Litter/Clutch size                  float64
Litters/Clutches per year           float64
Inter-litter/Interbirth interval    float64
Birth weight (g)                    float64
Weaning weight (g)                  float64
Adult weight (g)                    float64
Growth rate (1/days)                float64
Maximum longevity (yrs)             float64
Source                         

El dataset contiene 4513 especies y 31 variables. Las variables numéricas relacionadas con las medidas de historia de vida están almacenadas como float64, mientras que las variables de identificación y clasificación están almacenadas como texto. Además, Data quality permite diferenciar la confiabilidad de los registros y specimen origin permite distinguir entre especímenes silvestres y de cautiverio. Por otro lado, sample size es una variable categórica con cinco categorías, por lo que no representa directamente una cantidad numérica de individuos. La categoría más frecuente es medium, con 2408 registros. En general, antes de confiar en las medidas descriptivas es necesario considerar los valores faltantes, la calidad de los registros y el tamaño de muestra, ya que estas características pueden afectar la representatividad de los resultados.

**Punto 2**

In [17]:
numero_clases=df['Class'].nunique()
numero_clases

6

In [18]:
especie_clase = df['Class'].value_counts()
especie_clase

Class
Aves              1513
Mammalia          1349
Teleostei          806
Reptilia           547
Amphibia           181
Chondrichthyes     117
Name: count, dtype: int64

In [19]:
clase_mayor = especie_clase.idxmax()
n_especies_mayor = especie_clase.max()
clase_mayor, n_especies_mayor

('Aves', 1513)

**Punto 3**

In [ ]:
faltantes = (df.isnull().mean() * 100).sort_values(ascending=False)

print("Porcentaje de valores faltantes:")
print(faltantes)

print("\n5 con mayor porcentaje de datos faltantes:")
print(faltantes.head(5))

Porcentaje de valores faltantes:
MRDT (yrs)                          99.202304
IMR (per yr)                        99.113672
Weaning weight (g)                  91.491248
Temperature (K)                     89.053844
Growth rate (1/days)                87.613561
Metabolic rate (W)                  86.106803
Body mass (g)                       86.106803
Inter-litter/Interbirth interval    83.048970
Weaning (days)                      82.982495
Birth weight (g)                    72.745402
Litters/Clutches per year           71.105695
Gestation/Incubation (days)         59.384002
Male maturity (days)                58.342566
Litter/Clutch size                  53.667184
Female maturity (days)              45.512963
Adult weight (g)                    19.321959
Source                              12.275648
Maximum longevity (yrs)             10.813206
References                           0.022158
Common name                          0.000000
Order                                0.000000
F

Las cinco variables con mayor porcentaje de datos faltantes son MRDT (yrs)con 99.20%, IMR (per yr) con 99.11%,Weaning weight(g) con 91.49%, Temperature(K) con 89.05% y Growth rate con 87.61%. Una posible explicación es que estas variables requieren mediciones biológicas o experimentales más específicas, por lo que no se encuentran disponibles para todas las especies. En particular, variables como la tasa metabólica, temperatura crecimiento pueden ser difíciles de medir. Esto debe tenerse en cuenta porque realizar análisis sobre estas variables podría reducir considerablemente la cantidad de datos disponibles y afectar la representatividad de los resultados.

**Punto 4**

In [18]:
conteo_calidad = df['Data quality'].value_counts()

df_confiable = df[df['Data quality'].isin(['acceptable', 'high'])].copy()

n_total = len(df)
n_confiable = len(df_confiable)
porcentaje = (n_confiable / n_total) * 100

print(f"Porcentaje: {porcentaje:.2f}%")

Porcentaje: 86.95%


**Punto 5**

In [19]:
longevidad = df_confiable["Maximum longevity (yrs)"]

media = longevidad.mean()
mediana = longevidad.median()
moda = longevidad.mode()

print("Media:", media)
print("Mediana:", mediana)
print("Moda:", moda)

Media: 18.95223241590214
Mediana: 15.0
Moda: 0    10.0
Name: Maximum longevity (yrs), dtype: float64


La media de longevidad es de aproximadamente 18.95 años, mientras que la mediana es de 15 años y la moda es de 10 años. La media y la mediana sí son diferentes, ya que existe una diferencia de aproximadamente 3.95 años. La media es mayor que la mediana, lo que indica que algunos valores altos de longevidad pueden estar aumentando el promedio. Esto sugiere que la distribución de la longevidad presenta cierto sesgo hacia valores altos.

**Punto 6**

In [31]:
longevidad = df_confiable['Maximum longevity (yrs)']

rango = longevidad.max() - longevidad.min()
varianza = longevidad.var()
desv_est = longevidad.std()

rango, varianza, desv_est

(391.84, 292.145814948774, 17.0922735453413)

Estos estadísticos muestran que la longevidad varía mucho entre especies: el rango refleja la enorme distancia entre la especie más longeva y la más corta, pero depende solo de dos casos extremos, mientras que la varianza y la desviación estándar confirman que esa dispersión es alta en todo el conjunto y no un artefacto de uno o dos valores; en conjunto indican que no existe un "ritmo de vida" único en el reino animal, sino una distribución muy dispersa.

**Punto 7**

Para representar la longevidad típica de una especie, la medida mas adecuada seria la mediana, ya que cuando existen valores extremos que hacen que la media se aleje del como del centro de los datos, la mediana es menos sensible a valores muy muy altos o muy muy bajos, por lo que resulta más representativa, cuando la distribución de la longevidad  de una especie está sesgada a esos valores extremos que se pueden presentar.

**Punto 8**

In [34]:
longevidad_promedio = (
    df_confiable.groupby("Class")["Maximum longevity (yrs)"].mean().sort_values(ascending=False)
)

print(longevidad_promedio.round(2))

Class
Chondrichthyes    25.45
Reptilia          22.06
Mammalia          19.95
Aves              18.02
Teleostei         17.02
Amphibia          15.37
Name: Maximum longevity (yrs), dtype: float64


Chondrichthyes presenta la mayor longevidad promedio

Solo coincide en parte con la intuición. Que mamíferos y reptiles figuren entre los más longevos no sorprende, ya que incluyen ballenas, elefantes y tortugas gigantes y animales grandes. Pero que Chondrichthyes encabece la lista es menos intuitivo, porque uno tiende a pensar en tiburones como depredadores de vida "rápida". El resultado está dominado por unas pocas especies de aguas profundas y frías, lo que también se refleja en que esta clase tiene la desviación estándar más alta y muy pocas especie frente a Aves o Mammalia. Por eso el promedio no representa a un tiburón "típico", sino que está inflado por valores extremos, y conviene revisarlo junto con la mediana (que para Chondrichthyes baja a 19.25, mucho más cercana a Reptilia) antes de tomarlo como un patrón biológico confiable.

**Punto 9**

In [24]:
comparacion_origen = (
    df_confiable
    .groupby("Specimen origin")["Maximum longevity (yrs)"]
    .agg(["mean", "std"])
)
print(comparacion_origen)

                      mean        std
Specimen origin                      
captivity        20.472737  14.075964
unknown          17.870476  13.350734
wild             17.572471  19.575824


La mayor longevidad promedio se presenta en los animales de cautiverio, con aproximadamente 20.47 años. En comparación, los animales con origen desconocido tienen un promedio de 17.87 años, mientras que los animales en estado silvestre presentan el menor promedio, con 17.57 años. En cuanto a la variabilidad, el grupo silvestre tiene la desviación estándar más alta 19.58 años. En cambio, el cautiverio presenta una desviación estándar menor 14.08 años. Una posible explicación de la mayor longevidad en cautiverio es que los animales pueden tener alimentación controlada, atención veterinaria y menor exposición a depredadores. Sin embargo, estos resultados no permiten afirmar que el cautiverio sea directamente la causa de una mayor longevidad.

**Punto 10**

**Punto 11**

In [26]:
df_ordenado = df.sort_values(
    by="Maximum longevity (yrs)",
    ascending=False
)

df_ordenado[
    [
        "Species",
        "Common name",
        "Maximum longevity (yrs)",
        "Data quality",
        "Sample size"
    ]
].head(10)

,Species,Common name,Maximum longevity (yrs),Data quality,Sample size
1800,microcephalus,Greenland shark,392.0,acceptable,small
2196,mysticetus,Bowhead whale,211.0,acceptable,medium
4415,aleutianus,Rougheye rockfish,205.0,acceptable,medium
3684,nigra,Galapagos tortoise,177.0,acceptable,medium
4420,borealis,Shortraker rockfish,157.0,acceptable,medium
3679,gigantea,Aldabra tortoise,152.0,acceptable,large
3729,atlanticus,Orange roughy,149.0,acceptable,medium
4509,verrucosus,Warty oreo,140.0,acceptable,medium
3625,carolina,Eastern box turtle,138.0,acceptable,medium
3697,graeca,Mediterranean spur-thighed tortoise,127.0,acceptable,large


as diez especies con mayor longevidad presenta valores bastante altos, sobretodo el greenland shark con 392 años, seguido por la Bowhead whale con 211 años y el Rougheye rockfish con 205 años. Todos los registros tienen una calidad de datos aceptable, lo cual indica que son registros considerados confiables. Además, la mayoría tiene un tamaño de muestra medium, aunque algunas especies tienen muestras small o large. El caso del Greenland shark debe interpretarse con un poco más de cuidado, ya que aunque tiene una calidad acceptable, su tamaño de muestra es small. En general, los resultados muestran que existen especies con longevidades muy superiores al promedio del dataset, lo que también ayuda a explicar por qué la media (18.95 años) es mayor que la mediana (15 años).

**Punto 12**